client/landing/gutenboarding/gutenboard.tsx

```diff
@@ -66,9 +66,16 @@ export function Gutenboard() {
 		: undefined;
 	const goToPrevStep = currentStep > 0 ? () => setStep( step => step - 1 ) : undefined;
 
-	const onboardingBlock = useMemo( () => createBlock( name, { step: currentStep } ), [
-		currentStep,
-	] );
+	const onboardingBlock = useRef( createBlock( name, { step: currentStep } ) );
```

Comment:
Nice, haven't seen `ref`s used like this yet! :+1:

Reply:
Yeah, it's easy to reach for memo becuase you ostensibly want memoization, but memo is explicitly for perf purposes and has no lifetime guarantees. It's important to choose correctly:

> **You may rely on useMemo as a performance optimization, not as a semantic guarantee.** In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without `useMemo` — and then add it to optimize performance.
> […]
> The returned object [from `useRef`] will persist for the full lifetime of the component.

https://reactjs.org/docs/hooks-reference.html#usememo